integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class, NoH2.class, NoMySQLOld.class, NoOracle.class })
+    public void testJoinInlineEntityWithLimitWithCTE() {
+        Assume.assumeFalse(dbmsDialect.getLateralStyle() == LateralStyle.NONE);
+
+        doInJPA(entityManager -> {
+            QRecursiveEntity t = new QRecursiveEntity("t");
+            QRecursiveEntity subT = new QRecursiveEntity("subT");
+            QRecursiveEntity subT2 = new QRecursiveEntity("subT2");
+
+            List<Tuple> fetch = new BlazeJPAQuery<RecursiveEntity>(entityManager, cbf)
+                    .with(idHolderCTE, idHolderCTE.id)
+                    .as(select(document.id).from(document).where(document.id.eq(1L)))
+                    .select(t, subT2)
+                    .from(t)
+                    .leftJoin(select(subT).from(t.children, subT).orderBy(subT.id.asc()).limit(1), subT2)
+                    .where(t.id.in(select(idHolderCTE.id).from(idHolderCTE)))
+                    .lateral()
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class, NoMySQL.class })
+    public void testJoinInlineWithLimitUnion() {
+        doInJPA(entityManager -> {
+            QRecursiveEntity t = new QRecursiveEntity("t");
+            QRecursiveEntity subT = new QRecursiveEntity("subT");
+            QRecursiveEntity subT2 = new QRecursiveEntity("subT2");
+            QRecursiveEntity subT3 = new QRecursiveEntity("subT3");
+
+            JPQLQuery<RecursiveEntity> subA = select(
+                    new Binds<RecursiveEntity>()
+                        .bind(recursiveEntity.id, subT.id)
+                        .bind(recursiveEntity.name, subT.name)
+                        .bind(recursiveEntity.parent, subT.parent)
+            ).from(subT);
+
+            JPQLQuery<RecursiveEntity> subB = select(
+                    new Binds<RecursiveEntity>()
+                        .bind(recursiveEntity.id, subT3.id)
+                        .bind(recursiveEntity.name, subT3.name)
+                        .bind(recursiveEntity.parent, subT3.parent)).from(subT3);
+
+            SubQueryExpression<RecursiveEntity> union = new BlazeJPAQuery<RecursiveEntity>().unionAll(
+                    subA,
+                    subB
+            );
+
+            new BlazeJPAQuery<RecursiveEntity>(entityManager, cbf)
+                    .select(t, subT2)
+                    .from(t)
+                    .leftJoin(union, subT2).on(subT2.eq(subT2))
+                    .fetch();
+        });
+    }
+
+    @Test
+
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class, NoMySQL.class })
```

Comment:
Add a comment like e.g. `// NOTE: No advanced sql support for datanucleus, eclipselink and openjpa yet` and also `// NOTE: Entity joins are only supported on Hibernate 5.1+`

Also, why is MySQL excluded here?

Reply:
I copied the test profiles from the spring test suite, which still uses the old profiles for MySQL, so my tests with NoMySQLOld were not properly excluded. I hope this is a left over from my attempts figuring that one out :)